### ISSUE TITLE
'IPFS Desktop' can't be opened because Apple cannot check it for malicious software

### ISSUE THREAD

**Issue by brennanfife** [ipfs-shipyard/ipfs-desktop#1211]
Reinstalled Brave with the IPFS Chrome extension and the IPFS Companion Desktop application this morning. Opened it, but only get this warning with the following description: 'this software needs to be updated. Contact the developer for more information.' 

Currently running macOS Catalina 10.15

**Comment by pcowgill** [comment]
FYI this still is an issue for fresh installs

**Comment by andrew** [comment]
I ran into the same error this morning, it can be fixed by IPFS devs "Notarizing" the app with Apple before releasing it: https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution

**Comment by lidel** [comment]
Reopening. We need someone with Mac for this:

> Notarization **requires Xcode 10** or later. 
> Building a new app for notarization **requires macOS** 10.13.6 or later.

**Comment by hacdias** [comment]
Ref.: https://github.com/karaggeorge/electron-builder-notarize

**Comment by lidel** [comment]
No Mac at hands, but I moved the needle a bit:

- switched (https://github.com/ipfs-shipyard/ipfs-desktop/pull/1341) to the latest electron-builder to include fixes needed for notarization of mac artifact 
  - see "mac" changes [here](https://github.com/electron-userland/electron-builder/releases/tag/v22.3.1)
   - and [electron-notarize's prerequisites](https://github.com/electron/electron-notarize#prerequisites)

**Comment by lidel** [comment]
For now, I think we could try to manually notarize v0.10.3

Ref.
- IIUC we need someone with Xcode and AppleID
  - I am not sure if it needs to be the same ID as one used for signing:
    ```
    signing         file=dist/mac/IPFS Desktop.app identityName=Developer ID Application: Protocol Labs, Inc. (7Y229E2YRL) identityHash=C297E064280322D72444C869E7FC0F60B14E934E provisioningProfile=none
    ```
    @olizilla do you remember who owns keys used at Travis by any chance?
- Try executing step (2) described at [_Notarize Your Preexisting Software_](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution#3087722), but note that:
  > Alternatively, you can put apps, kernel extensions, and other software in a container, like a disk image, and notarize the container. The notary service accepts disk images (UDIF format), signed flat installer packages, and ZIP archives.
  - So instead of paclaging a local project, try to upload existing Disk Image `.dmg`: 
    -  [IPFS-Desktop-0.10.3.dmg](https://github.com/ipfs-shipyard/ipfs-desktop/releases/download/v0.10.3/IPFS-Desktop-0.10.3.dmg) → [Upload Your App to the Notarization Service](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow#3087734)
   - Step (3) is optional, (2) should be enough for online notarization checks to pass (most of the users will download `.dmg` from the internet anyway)

**Comment by lidel** [comment]
Thanks to help from @olizilla and @autonome  this is now fixed in [0.10.4](https://github.com/ipfs-shipyard/ipfs-desktop/releases/tag/v0.10.4) \o/


### 0.10.3

> <img width="532" alt="Screenshot 2020-03-12 at 14 18 07" src="https://user-images.githubusercontent.com/58871/76531071-8fcc5500-646c-11ea-9c6c-d4f8755d80ad.png">
 
### 0.10.4

> <img width="593" alt="Screenshot 2020-03-12 at 14 18 30" src="https://user-images.githubusercontent.com/58871/76531116-a1adf800-646c-11ea-8699-3ec53af34563.png">

Details: https://github.com/ipfs-shipyard/ipfs-desktop/pull/1365

**Comment by slhodak** [comment]
I get the "Is an app downloaded from the internet" warning, and click open, and Activity Monitor says the following processes are running, but there is no ipfs window, nor a dot next to the icon in the dock.
ipfs
IPFS Desktop
IPFS Desktop Helper (Renderer)
IPFS Desktop Helper (GPU)

## System
Catalina 10.15.7

I do have two graphics card (one additional) but am not sure if that's related:
NVIDIA GeForce GT 650M 512 MB
Intel HD Graphics 4000 1536 MB

And I only have 35GB of free disk space--would that be an issue?

**Comment by slhodak** [comment]
After trying it out, I think I should mention here that I've run IPFS from the command line and it seems to work fine, including the local browser UI.

**Comment by anthraxblast** [comment]
I'm getting the same issue as slhodak - no matter what I do, I cannot get IPFS desktop to work on Mac.  MacBook Air (13-inch, Early 2015), 1.6 GHz Dual-Core Intel Core i5, 8 Gig Memory.  MacOS, Big Sur v.11.4.  The error I get on an administrator account:
"You do not have permission to open the application “IPFS Desktop”.  Contact your computer or network administrator for assistance."  Note that I got an earlier version of IPFS to work on the prior OS before Big Sur (which was Catalina/Jazz), but it broke with Big Sur.  I tried putting it in the Applications folder and using the CTRL key to open it, per the docs - no help.  Bit surprised to see a desktop client fail like this - I download all sorts of smallish 3rd party apps that work fine.
